### PR TITLE
Add a valueForUndefinedKey

### PR DESCRIPTION
**Pull request by vvanchesa**
To ignore unknown parameters in a RKPaginatorObject. A URL my already have a colon in it to represent a custom reference to a API object. This pull request will ignore such unknown parameters. 
A URL like /elements/DO:1234/Structure will now work with. Without the change, the URL will be parsed as a patterned URL and will fail.

### CODE REVIEW

**Comment by valeriomazzeo** [comment]
can you please write a test for it?

**Comment by vvanchesa** [comment]
OK, 
I locally cloned the rep and followed the instructions on https://github.com/vvanchesa/RestKit/tree/development/Tests

I started the rake server and run the tests.. Got 284 errors? 


To test my pull request (Allowing colons in URL)  I assume to add a new route in the server.rb file: 

get '/paginate/doc:123/' do
    status 200
    content_type 'application/json'
    {:per_page => 10, :total_entries => 0,
        :current_page => 1, :entries => [], :total_pages => 0}.to_json
  end


and in RKPaginatorTest a new test: 

- (void)testPaginatorWithPaginationURLThatIncludesAColon
{
    NSURL *paginationURL = [NSURL URLWithString:@"/paginate/do:123/?per_page=:perPage&page=:currentPage" relativeToURL:[RKTestFactory baseURL]];
    NSURLRequest *request = [NSURLRequest requestWithURL:paginationURL];
    RKPaginator *paginator = [[RKPaginator alloc] initWithRequest:request paginationMapping:self.paginationMapping responseDescriptors:@[ self.responseDescriptor ]];
    __block NSArray *blockObjects = nil;
    [paginator setCompletionBlockWithSuccess:^(RKPaginator *paginator, NSArray *objects, NSUInteger page) {
        blockObjects = objects;
    } failure:nil];
    [paginator loadPage:1];
    [paginator waitUntilFinished];
    expect(blockObjects).willNot.beNil();
    expect(paginator.pageCount).to.equal(0);
    expect(paginator.objectCount).to.equal(0);
}


What did I wrong that the defaults Tests are mostly failing?

**Comment by valeriomazzeo** [comment]
did you try revert back your changes and see if all the tests passes?

**Comment by vvanchesa** [comment]
The bunch of error occurs in different classes - they have nothing to do with the changes I made. 
I also reverted my changes back. 


the Xcode Test and in the command line a ’rake test’ results to a bunch like this: 
✗ testDeletionOfOrphanedTagsOfPosts, expected: nil/null, got: Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo=0x7f81609c9030 {NSUnderlyingError=0x7f81602d86e0 "Could not connect to the server.", NSErrorFailingURLStringKey=http://127.0.0.1:4567/posts.json, NSErrorFailingURLKey=http://127.0.0.1:4567/posts.json, _kCFStreamErrorDomainKey=1, _kCFStreamErrorCodeKey=61, NSLocalizedDescription=Could not connect to the server.}

**Comment by valeriomazzeo** [comment]
it looks like you didn't start the server (although I am sure you did), maybe you have something else running on that same port (4567)?

**Comment by vvanchesa** [comment]
I also tried to start only ’rake test’ with- and without running rake server in a different terminal. 

Sever is up and running - and even respond to (some?) requests: 
thorstens-mac:Tests thorstenclaus$ rake server
(in /Users/thorstenclaus/Documents/Develop/integration/RestKit)
Executing: `bundle exec rackup -s thin -o localhost -p 4567 -P Tests/Server/server.pid Tests/Server/server.ru`
>> Thin web server (v1.5.1 codename Straight Razor)
>> Maximum connections set to 1024
>> Listening on localhost:4567, CTRL+C to stop
::1 - - [24/May/2016 12:30:55] "GET /object_manager/1234/cancel HTTP/1.1" 204 - 0.0640

The last line appears when running the tests. 

To make a long story short: 
May I simply commit my changes and commit a test „blind“ and make again the pull request. Maybe travis has a different configuration and will pass the tests?

**Comment by valeriomazzeo** [comment]
you can, but it will take longer to test it with travis everytime if anything goes wrong

**Comment by vvanchesa** [comment]
There is a difference: 

In Gemfile.Lock: 

-    eventmachine (1.0.7)
+    eventmachine (1.2.0.1)

and

 BUNDLED WITH
-   1.10.6
+   1.12.4


Maybe this has something to do with the failing tests?

File: Gemfile.lock [hunk 1]
```diff
@@ -102,4 +102,4 @@ DEPENDENCIES
   xctasks (~> 0.5.0)
 
 BUNDLED WITH
-   1.10.6
+   1.12.4
```
Comment (valeriomazzeo) [inline Gemfile.lock]: can you please revert this changes?

Comment (vvanchesa) [inline Gemfile.lock]: I reverted these changes in the Lock file and Tests now run gracefully.
New Test is called „testPaginatorWithPaginationURLThatIncludesAColon“ and also succeeds.

**Comment by valeriomazzeo** [comment]
it looks ok to me 👍
let's wait for @segiddins to have a look too.

**Comment by segiddins** [comment]
This is kinda gross -- this will break things for anyone who's actually using KVC on the paginator

**Comment by vvanchesa** [comment]
The KVC is a very good point. So simply cancel my pull request in this case. I will fix the issue by locally make a category definition for the -

valueForUndefinedKey


Thanks for all

**Comment by valeriomazzeo** [comment]
hey @vvanchesa I have an idea.

`RKPaginator` is internally using `SOCKit` for pattern matching.

Did you try escaping the colon in your path using `\\`?

In your example it should look like: `/elements/DO\\:1234/`

I think it should work

**Comment by vvanchesa** [comment]
Yes, I did. 
Even tried the Unicode. 
In Case of the escaping like //: somewhere the url can not be build. 
In case of the Unicode kvc tries to find s method '1234' in the pagination class. 

My solution is now simply added a category file in my project and implemented there the 'valueForUndefinedKey' method.
